lib/AST/ASTImporter.cpp

```diff
@@ -6081,6 +6082,31 @@ Expr *ASTNodeImporter::VisitUnaryExprOrTypeTraitExpr(
                                           Importer.Import(E->getRParenLoc()));
 }
 
+Expr *ASTNodeImporter::VisitTypeTraitExpr(TypeTraitExpr *E) {
+  QualType T = Importer.Import(E->getType());
+  if (T.isNull())
+    return nullptr;
+
+  unsigned NumArgs = E->getNumArgs();
+  llvm::SmallVector<TypeSourceInfo *, 2> ToArgTypes(NumArgs);
+  for (unsigned AI = 0, AE = NumArgs; AI != AE; ++AI) {
+    TypeSourceInfo *FromArgType = E->getArg(AI);
+    TypeSourceInfo *ToArgType = Importer.Import(FromArgType);
+    if (!ToArgType)
+      return nullptr;
+    ToArgTypes[AI] = ToArgType;
+  }
+  TypeSourceInfo **ToArgTypesArray = new (Importer.getToContext())
+      TypeSourceInfo*[NumArgs];
+  for (unsigned AI = 0, AE = NumArgs; AI != AE; ++AI)
```

Comment:
While running the first loop, it can turn out that we bail out. Because we don't know how often it happens it is better not to trash the memory with needless new delete pairs and just allocate the thing on stack instead.